File: 0001-0500/0217_containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()
        for num in nums:
            if num not in hashset:
                hashset.add(num)
            else:
                return True
        return False

# Alternative solution

class Solution1:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))

File: 0001-0500/0217_containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num: nums) {
            if (set.contains(num)) return true;
            set.add(num);
        }
        return false;
    }
}

// Alternative solution

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = Arrays.stream(nums).boxed().collect(Collectors.toCollection(HashSet::new));
        return nums.length != set.size();
    }
}
